### PR TITLE
MSPSDS-498 record case comms email

### PR DESCRIPTION
**Pull request by MatthewBurstein**
Not ready for merging yet but would be interested to get feedback. The bulk of the functionality is in place but there is definitely a lot of room for improvement around adding multiple attachments, especially in FileConcern, EmailsController and AuditActivity::Correspondence::AddEmail and there are some fiddly issues to fix also.

Should we have a separate activity display partial for each each kind of activity in the activities summary? Should larger refactoring like this be a separate PR?

### CODE REVIEW

File: web/app/assets/javascripts/investigations/correspondence.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+$(document).on('turbolinks:load', function () {
+  $('#correspondence_email_attachment').on('change', function () {
+    var attachementDescription = $('#attachement-description');
```
Comment (tom-bell-softwire) [inline web/app/assets/javascripts/investigations/correspondence.js]: This bit of functionality (attachment description field shown only after selecting a file) appears on a number of activity pages, so I'm planning to pull it out into a partial and associated JS file.

Depending on when that gets merged in, we can then remove this.

File: web/app/controllers/investigations/emails_controller.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    initialize_file_attachment :email_file
+    initialize_file_attachment :email_attachment
+    redirect_to wizard_path(steps.first, request.query_parameters)
+  end
+
+  def create
+    attach_file_to_list(@email_file, :email_file, @correspondence.documents)
+    attach_file_to_list(@email_attachment, :email_attachment, @correspondence.documents)
+    @investigation.correspondences << @correspondence
+    @investigation.save
```
Comment (tom-bell-softwire) [inline web/app/controllers/investigations/emails_controller.rb]: Do we need a check here to handle a failed `save` due to validation errors? Perhaps unlikely when following the wizard steps, but a direct POST request (i.e. API call) could contain invalid properties.

File: web/app/controllers/investigations/emails_controller.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    AuditActivity::Correspondence::AddEmail.from(@correspondence, @investigation)
+    redirect_to investigation_path(@investigation)
```
Comment (tom-bell-softwire) [inline web/app/controllers/investigations/emails_controller.rb]: I think the convention elsewhere is to include a flash success notice when redirecting to the case page. Is that explicitly _not_ wanted here?

File: web/db/migrate/20181109145423_add_email_subject_direction_and_attachement_description_to_correspondence.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class AddEmailSubjectDirectionAndAttachementDescriptionToCorrespondence < ActiveRecord::Migration[5.2]
+  def change
+    safety_assured do
+      change_table :correspondences do |t|
+        t.string "email_direction"
+        t.text "attachment_description"
```
Comment (tom-bell-softwire) [inline web/db/migrate/20181109145423_add_email_subject_direction_and_attachement_description_to_correspondence.rb]: Should we be storing the attachment description as a field on the correspondence? I think it should just be set in the metadata of the relevant attachment blob.

File: web/app/assets/javascripts/investigations/correspondence.js (hunk 1 above)
Comment (MatthewBurstein) [reply web/app/assets/javascripts/investigations/correspondence.js]: Should I wait for the merge to see what you have done before working on this?

Comment (tom-bell-softwire) [reply web/app/assets/javascripts/investigations/correspondence.js]: Unless you're planning a major refactor of this, no need to wait for the other branch to be merged! Should be easy to refactor this afterwards instead.

File: web/db/schema.rb [hunk 5]
```diff
@@ -10,10 +10,9 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2018_11_08_172853) do
+ActiveRecord::Schema.define(version: 2018_11_09_145423) do
```
Comment (MatthewBurstein) [inline web/db/schema.rb]: The version number causes a merge conflict with master. What's the best way to resolve this?

Comment (broder) [reply web/db/schema.rb]: Delete the file and run `db:migrate` again locally

File: web/app/assets/javascripts/investigations/correspondence.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+$(document).on('turbolinks:load', function () {
```
Comment (broder) [inline web/app/assets/javascripts/investigations/correspondence.js]: Do we need this as well as `attachment-description.js`?

File: web/db/schema.rb [hunk 7]
```diff
@@ -164,6 +166,26 @@
     t.index ["assignee_id"], name: "index_investigations_on_assignee_id"
   end
 
+  create_table "pghero_query_stats", force: :cascade do |t|
```
Comment (broder) [inline web/db/schema.rb]: We removed these recently - they shouldn't be in `schema.rb` anymore

Comment (MatthewBurstein) [reply web/db/schema.rb]: I deleted the schema and migrated again but they are still there. How should I get rid?

File: web/app/assets/javascripts/investigations/correspondence.js (hunk 6 above)
Comment (MatthewBurstein) [reply web/app/assets/javascripts/investigations/correspondence.js]: Yes meant to do this. Have deleted

**Pull request merged**